### PR TITLE
e2e: topomgr: remove policy check

### PR DESCRIPTION
**Pull request by fromanirh**
When we introduced the topology manager extended tests, we added a check
to ensure that the topology manager was configured with the
`single-numa-node` policy. This because the `single-numa-node` policy
is the strictest one, so this is the simplest scenario on which we can
check the correct outcome of the allocation.

Checking other policies in the general case requires either
- complex testing code to properly detect various legal resource allocation
- accept the chance of having false negatives (test code misdetecting legal allocations)
- even a stricter coupling between test code and production code

However, we want to be able to check other policies.
The simplest possible step forward is to just remove the check for the
policy and to put on who sets the test up the responsability of
configuring the environment on such a way that test is relevant.
In other words, that could mean to over-provision the environment to
make sure the aligned allocations are possible with less strict
policies.

This change nothing for the current test scenarios, on which we set the
single-numa-node policy anyway, but makes it possible to use the
testsuite against clusters configured with the `best-effort` policy,
like it is the case when the `performance-addon-operator` is enabled
in the cluster.

Signed-off-by: Francesco Romani <fromani@redhat.com>

### CODE REVIEW

**Comment by fromanirh** [comment]
/cc @vpickard @Deepthidharwar 
this is the change we discussed

**Comment by fromanirh** [comment]
This PR is reviewable. The tests still work as expected against a cluster (disclosure: cluster configured with `single-numa-node` policy but for verification's sake each policy is equally good)

**Comment by fromanirh** [comment]
checking why CI complains

**Comment by fromanirh** [comment]
@sjenning could you please review?

**Comment by fromanirh** [comment]
obsolete. Let's re-evaluate later on.
